### PR TITLE
add operator upgrade case

### PR DESCRIPTION
**Pull request by weekface**
### What problem does this PR solve? 

- add operator upgrade case

### Check List 

Tests 

 - E2E test
 - Stability test

Code changes

 - Has Go code change

Side effects

Related changes

### Does this PR introduce a user-facing change?:
 
 ```release-note
NONE
 ```

### CODE REVIEW

File: tests/failover.go [hunk 1]
```diff
@@ -106,6 +108,30 @@ func (oa *operatorActions) TruncateSSTFileThenCheckFailover(info *TidbClusterCon
...
+	}
+
+	glog.Infof("deleting pod: [%s/%s] again", info.Namespace, store.PodName)
+	return wait.Poll(10*time.Second, time.Minute, func() (bool, error) {
+		err = oa.kubeCli.CoreV1().Pods(info.Namespace).Delete(store.PodName, &metav1.DeleteOptions{})
+		if err != nil {
+			return false, nil
+		}
+		return true, nil
+	})
```
Comment (xiaojingchen) [inline tests/failover.go]: check the pod is recreated?

Comment (weekface) [reply tests/failover.go]: yes, we have check it recreated in later logic

**Comment by weekface** [comment]
@cofyc PTAL

File: tests/slack/slack.go [hunk 2]
```diff
@@ -11,8 +11,9 @@ import (
 )
 
 var (
-	Channel    string
-	WebhookURL string
+	Channel      string
+	WebhookURL   string
+	SuccessCount int
```
Comment (cofyc) [inline tests/slack/slack.go]: `SuccessCount` is not used?

Comment (weekface) [reply tests/slack/slack.go]: It is used at these postions:

https://github.com/pingcap/tidb-operator/pull/579/files#diff-2ef72aa70c90226cd8aac8a436f9bff9R150
https://github.com/pingcap/tidb-operator/pull/579/files#diff-8f8eafaeec65b7926966885b1c103ca8R58
https://github.com/pingcap/tidb-operator/pull/579/files#diff-8f8eafaeec65b7926966885b1c103ca8R300

**Pull request merged**